### PR TITLE
clusterloader2: Make sure the latency is not a negtive value

### PR DESCRIPTION
**Pull request by chendave**
fix: https://github.com/kubernetes/perf-tests/issues/1307
Signed-off-by: Dave Chen <dave.chen@arm.com>

### CODE REVIEW

File: clusterloader2/pkg/measurement/util/phase_latency.go [hunk 1]
```diff
@@ -102,7 +102,17 @@ func (o *ObjectTransitionTimes) CalculateTransitionsLatency(t map[string]Transit
 				klog.V(4).Infof("%s: failed to find %v time for %v", o.name, transition.To, key)
 				continue
 			}
-			lag = append(lag, latencyData{key: key, latency: toPhaseTime.Sub(fromPhaseTime)})
+			latencyTime := toPhaseTime.Sub(fromPhaseTime)
+			// latencyTime should be always larger than zero, however, in some cases, the precision of timestamp
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/util/phase_latency.go:106]: can you please shorten the lines in this comment?

Comment (chendave) [reply clusterloader2/pkg/measurement/util/phase_latency.go]: sure, could you pls take another look?

File: clusterloader2/pkg/measurement/util/phase_latency.go [hunk 2]
```diff
@@ -102,7 +102,14 @@ func (o *ObjectTransitionTimes) CalculateTransitionsLatency(t map[string]Transit
 				klog.V(4).Infof("%s: failed to find %v time for %v", o.name, transition.To, key)
 				continue
 			}
-			lag = append(lag, latencyData{key: key, latency: toPhaseTime.Sub(fromPhaseTime)})
+			latencyTime := toPhaseTime.Sub(fromPhaseTime)
+			// latencyTime should be always larger than zero, however, in some cases, it might be a negative value
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/util/phase_latency.go:106]: Sorry - apparently I wasn't clear enough.
I didn't mean shortening the comment itself (though the current version is also fine).

I meant: making the lines shorter (i.e. splitting the lines earlier than you do now)

File: clusterloader2/pkg/measurement/util/phase_latency.go [hunk 3]
```diff
@@ -102,7 +102,14 @@ func (o *ObjectTransitionTimes) CalculateTransitionsLatency(t map[string]Transit
...
+			// due to the precision of timestamp can only get to the level of second, the microsecond and nanosecond
+			// have been discarded purposely in kubelet because apisever does not support RFC339NANO.
```
Comment (chendave) [inline clusterloader2/pkg/measurement/util/phase_latency.go:108]: typo, s/apisever/apiserver

File: clusterloader2/pkg/measurement/util/phase_latency.go (hunk 2 above)
Comment (chendave) [reply clusterloader2/pkg/measurement/util/phase_latency.go]: should be fine now.

File: clusterloader2/pkg/measurement/util/phase_latency.go (hunk 3 above)
Comment (chendave) [reply clusterloader2/pkg/measurement/util/phase_latency.go]: done

**Comment by wojtek-t** [comment]
/lgtm
/approve

Thanks!

**Comment by mm4tt** [comment]
/lgtm

Thanks!

**Pull request merged**